docs/managedidentity_public_api.md

```diff
@@ -0,0 +1,206 @@
+
+```go
+import (
+    "context"
+    "fmt"
+    "net/http"
+
+    mi "github.com/AzureAD/microsoft-authentication-library-for-go/apps/managedidentity"
+)
+
+func RunManagedIdentity() {
+    customHttpClient := &http.Client{}
+
+    miSystemAssigned, error := mi.New(mi.SystemAssigned())
+    if error != nil {
+        fmt.Println(error)
+    }
+
+    miClientIdAssigned, error := mi.New(mi.ClientID("client id 123"), mi.WithHTTPClient(customHttpClient))
+    if error != nil {
+        fmt.Println(error)
+    }
+
+    miResourceIdAssigned, error := mi.New(mi.ResourceID("resource id 123"))
+    if error != nil {
+        fmt.Println(error)
+    }
+
+    miObjectIdAssigned, error := mi.New(mi.ObjectID("object id 123"))
+    if error != nil {
+        fmt.Println(error)
+    }
+
+    miSystemAssigned.AcquireToken(context.Background(), "resource", mi.WithClaims("claim"))
+
+    miClientIdAssigned.AcquireToken(context.Background(), "resource")
+
+    miResourceIdAssigned.AcquireToken(context.Background(), "resource", mi.WithClaims("claim"))
+    
+    miObjectIdAssigned.AcquireToken(context.Background(), "resource")
+}
+```
+
+To create a new **ManagedIdentityClient**
+
+```go
+// Client to be used to acquire tokens for managed identity.
+// ID: [SystemAssigned()], [ClientID("clientID")], [ResourceID("resourceID")], [ObjectID("objectID")]
+//
+// Options: [WithHTTPClient]
+func New(id ID, options ...Option) (Client, error) {
+    // implementation details
+}
+```
+
+The options available for passing to the client are  
+
+```go
+// WithHTTPClient allows for a custom HTTP client to be set.
+func WithHTTPClient(httpClient ops.HTTPClient) Option {
+    // implementation details
+}
+```
+
+The options available for the request are 
+
+```go
+// WithClaims sets additional claims to request for the token, such as those required by conditional access policies.
+// Use this option when Azure AD returned a claims challenge for a prior request. The argument must be decoded.
+func WithClaims(claims string) AcquireTokenOption {
+    // implementation details
+}
+```
+
+## Error Handling
+
+Error handling in GO is different to what we used to in languages like Java or Swift.
+There is no concept of ‘exceptions’, instead we just return errors and immediately check if an error was returned and handle it there and then.  
+Basically, we will return client-side errors like so:
+
+```go
+if err != nil {
+    return errors.New("Some Managed Identity Error here”) 
+}
+```
+
+This will be inside of any client methods that throw errors, using descriptive errors based on the .NET and Java Implementation. These errors will be propagated down the chain and handled when they are received
+
+For service side errors it works a little differently
```

Comment:
The service errors need to include the MSI Source. This is a requirement from Azure SDK, but also to help the team with supportability.

Reply:
Yeah as you mentioned on the public API document, service errors will be returned and handled differently.
Is what is mentioned in the document not okay and you want it changed or are you just noting this here?

Reply:
The service errors from MSI have to follow the same pattern as the service error from AAD. In addition to that, the Azure source (VM, Arc etc.) should be clear in the errors (e.g. in the message), to help troubleshooting.

Reply:
Yep, I see in Java they return a string which is getting the error response, description, correlation id etc, so we will do the same.
This is just copied from the public api doc, is there something more that you would like me to add here?